scripts/skills/perks/perk_rf_phalanx.nut

```diff
@@ -45,45 +51,40 @@ this.perk_rf_phalanx <- ::inherit("scripts/skills/skill", {
 
 	function onAnySkillUsed( _skill, _targetEntity, _properties )
 	{
-		if (_targetEntity != null && _skill.isAttack() && !_skill.isRanged())
+		if (_skill.isAttack() && !_skill.isRanged() && (this.m.RequiredDamageType == null || _skill.getDamageType().contains(this.m.RequiredDamageType)))
 		{
-			_properties.Reach += this.getCount();
+			_properties.Reach += this.getAllyCount();
 		}
 	}
 
 	function onBeingAttacked( _attacker, _skill, _properties )
 	{
 		if (_skill.isAttack() && !_skill.isRanged())
 		{
-			_properties.Reach += this.getCount();
+			_properties.Reach += this.getAllyCount();
 		}
 	}
 
-	function onAfterUpdate( _properties )
+	function onUpdate( _properties )
 	{
-		if (this.hasAdjacentShieldwall())
+		if (this.isActorValid(this.getContainer().getActor()))
```

Comment:
I suggest adding a new `isEnabled()` function, which just calls `this.isActorValid(this.getContainer().getActor())` internally.
It does not safe much currently except making the code a bit more readable and it follows our growing standardization in skills, where such a function usually mirrors the requirement line in perks.

But in the near future we might divorce again the conditions for the perk being enabled on the user, and the conditions for an adjacent ally being eligable for counting. Then it will come in handy